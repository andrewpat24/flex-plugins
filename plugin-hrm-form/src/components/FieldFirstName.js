import React from 'react';
import { StyledInput,
         StyledLabel,
         ErrorText,
         TextField
} from '../Styles/HrmStyles';

      //     <TextField>
      //     <StyledLabel>First name</StyledLabel>
      //     <StyledInput theme={this.props.theme} 
      //       name='firstName'
      //       value={this.props.form.callerInformation.name.firstName}
      //       onChange={(e) => 
      //         this.props.handleChange(taskId,
      //                                 ['callerInformation', 'name'],
      //                                 e)}
      //     />
      // </TextField>
      // error={true}
      // <ErrorText>Required field</ErrorText>

const FieldFirstName = (props) => {
  return (
    <TextField>
      <StyledLabel>First Name</StyledLabel>
      <StyledInput
        error={props.form.callerInformation.name.firstName.error !== null}
        name='firstName'
        value={props.form.callerInformation.name.firstName.value}
        onChange={(e) =>
          props.handleChange(props.taskId,
                             ['callerInformation', 'name'],
                             e)}
      />
      {props.form.callerInformation.name.firstName.error ?
        <ErrorText>{props.form.callerInformation.name.firstName.error}</ErrorText> :
        ''
      }
    </TextField>
  );
};

export default FieldFirstName;
