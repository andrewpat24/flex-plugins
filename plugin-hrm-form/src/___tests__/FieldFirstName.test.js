jest.mock('../Styles/HrmStyles', () => {
  return {
    StyledInput: 'StyledInput',
    StyledLabel: 'StyledLabel',
    ErrorText: 'ErrorText',
    TextField: 'TextField'
  };
});
import React from 'react';
import FieldFirstName from '../components/FieldFirstName';
import renderer from 'react-test-renderer';

test('FieldFirstName renders initially', () => {
  const form = {
    callerInformation: {
      name: {
        firstName: {
          value: '',
          touched: false,
          error: null
        }
      }
    }
  };
  const handleChange = jest.fn();
  const taskId = '';
  const component = renderer.create(
    <FieldFirstName
      form={form}
      handleChange={handleChange}
      taskId={taskId}
    />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('FieldFirstName renders when valid', () => {
  const form = {
    callerInformation: {
      name: {
        firstName: {
          value: 'testValue',
          touched: false,
          error: null
        }
      }
    }
  };
  const handleChange = jest.fn();
  const taskId = '';
  const component = renderer.create(
    <FieldFirstName
      form={form}
      handleChange={handleChange}
      taskId={taskId}
    />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('FieldFirstName renders errors when not valid', () => {
  const form = {
    callerInformation: {
      name: {
        firstName: {
          value: 'testValue',
          touched: true,
          error: 'This field is required'
        }
      }
    }
  };
  const handleChange = jest.fn();
  const taskId = '';
  const component = renderer.create(
    <FieldFirstName
      form={form}
      handleChange={handleChange}
      taskId={taskId}
    />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
