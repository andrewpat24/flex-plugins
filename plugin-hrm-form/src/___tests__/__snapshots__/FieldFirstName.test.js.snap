// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FieldFirstName renders errors when not valid 1`] = `
<TextField>
  <StyledLabel>
    First Name
  </StyledLabel>
  <StyledInput
    error={true}
    name="firstName"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    value="testValue"
  />
  <ErrorText>
    This field is required
  </ErrorText>
</TextField>
`;

exports[`FieldFirstName renders initially 1`] = `
<TextField>
  <StyledLabel>
    First Name
  </StyledLabel>
  <StyledInput
    error={false}
    name="firstName"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    value=""
  />
  
</TextField>
`;

exports[`FieldFirstName renders when valid 1`] = `
<TextField>
  <StyledLabel>
    First Name
  </StyledLabel>
  <StyledInput
    error={false}
    name="firstName"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    value="testValue"
  />
  
</TextField>
`;
